sdk/test-utils/recorder/GUIDELINES.md

```diff
@@ -0,0 +1,322 @@
+# GUIDELINES FOR RECORD AND PLAYBACK
+
+## Setup for record and playback
+
+**New env variable for recordings - TEST_MODE** [Supposed to be added in the environment(or `.env` file if the package uses `dotenv` as a devDependency) to be able to do record and playback]
+
+- If TEST_MODE = "record",
+  - Tests hit the live-service
+  - Nock/Nise are leveraged for recording the request-responses for future use
+  - If recordings are already present, forces re-recording
+- Else If TEST_MODE = "playback",
+  - Existing recordings are used
+- Else
+  - Tests hit the live-service, we don't record the requests/responses
+
+Add `@azure/test-utils-recorder` as a devDependency of your sdk.
+
+## Adding a new test/test-suite
+
+- `recorder` package assumes that the tests in the sdk are leveraging [mocha](https://mochajs.org/) and [rollup](https://rollupjs.org/guide/en/) (and [karma](https://karma-runner.github.io/latest/index.html) test runner for browser tests) as suggested by the [template](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/template/template) package in the repo.
+
+- Follow the below template for adding a new test. `before` and `after` sections are optional, `beforeEach` and `afterEach` sections are compulsory.
```

Comment:
Is this correct? I believe it can be either. If I decide to just use before and after, I think I'm able to skip beforeEach and afterEach, same as if I decide to go with the *Each functions, then before and after are optional. At least that's how I've been doing things in keyvault.

Reply:
[Discussed offline with @sadasant]
Decided to follow "starting the recorder in beforeeach and stopping in aftereach" method since it would allow recording a specific test separately.
I'll rephrase the statement in the Guidelines explaining how the recorder starts and stops which would clarify the need for beforeEach and afterEach sections.